platform/mds/mds-web/src/main/java/org/motechproject/mds/web/service/impl/InstanceServiceImpl.java

```diff
@@ -103,7 +103,7 @@ public Object saveInstance(EntityRecord entityRecord) {
     public Object saveInstance(EntityRecord entityRecord, Long deleteValueFieldId) {
         EntityDto entity = getEntity(entityRecord.getEntitySchemaId());
         validateCredentials(entity);
-        validateNonEditableProperty(entity);
+        validateNonEditableProperty(entity.getId());
```

Comment:
Now there will be two queries to the database instead of one query, maybe we should overload this method.